Makefile

```diff
@@ -1758,7 +1756,7 @@ fuzz: bin/fuzz
 bazel-generate: ## Generate all bazel BUILD files.
 	@echo 'Generating DEPS.bzl and BUILD files using gazelle'
 	@bazel run //:gazelle -- update-repos -from_file=go.mod -build_file_proto_mode=disable -to_macro=DEPS.bzl%go_deps
-	@bazel run //:gazelle
+	@bazel run //:gazelle -- -index=false
```

Comment:
Hmmm... I don't love this change. The `index` functionality was really useful for me with the Protobuf PR.

Why is this necessary? I'd like to avoid this if at all possible. I think if it's absolutely required we can keep it and just work around the deficiencies from the lack of an `index` in follow-up PRs, though it'll probably make our lives harder.

Reply:
Then again, maybe the fact that we're not vendoring anything makes `-index=false` that much easier to work with. Can't really say for sure, honestly.

Reply:
This is actually the very thing that forces gazelle to generate external targets. See https://github.com/bazelbuild/bazel-gazelle#dependency-resolution. Given we usually have `vendor` checked out locally, we want it to _not_ resolve dependencies to their vendored paths. Not having this lets gazelle index the entire repo first, and has it default to using our previous paths. Even forcing `external = "external"` is not enough to override that behavior.